ruleset/rules/0750-github_rules.xml

```diff
@@ -0,0 +1,3225 @@
+    </rule>
+
+    <!-- environment
+        create_actions_secret
+        delete
+        remove_actions_secret
+        update_actions_secret
+        add_protection_rule
+        update_protection_rule
+        remove_protection_rule
+    -->
+    <rule id="91149" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^environment.\.*</action>
+        <description>GitHub Environment.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.create_actions_secret -->
+    <rule id="91150" level="7">
+        <if_sid>91149</if_sid>
+        <action>environment.create_actions_secret</action>
+        <description>GitHub Environment Create actions secret.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.delete -->
+    <rule id="91151" level="7">
+        <if_sid>91149</if_sid>
+        <action>environment.delete</action>
+        <description>GitHub Environment Delete.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.remove_actions_secret -->
+    <rule id="91152" level="9">
+        <if_sid>91149</if_sid>
+        <action>environment.remove_actions_secret</action>
+        <description>GitHub Environment Remove actions secret.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.update_actions_secret -->
+    <rule id="91153" level="7">
+        <if_sid>91149</if_sid>
+        <action>environment.update_actions_secret</action>
+        <description>GitHub Environment Update actions secret.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.add_protection_rule -->
+    <rule id="91154" level="7">
+        <if_sid>91149</if_sid>
+        <action>environment.add_protection_rule</action>
+        <description>GitHub Environment Add protection rule.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.update_protection_rule -->
+    <rule id="91155" level="7">
+        <if_sid>91149</if_sid>
+        <action>environment.update_protection_rule</action>
+        <description>GitHub Environment Update protection rule.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!--  environment.remove_protection_rule -->
+    <rule id="91156" level="9">
+        <if_sid>91149</if_sid>
+        <action>environment.remove_protection_rule</action>
+        <description>GitHub Environment Remove protection rule.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!-- git
+            clone
+            fetch
+            push
+    -->
```

Comment:
Does this mean that an alert will be generated for each clone, fetch, and push? Perhaps level should be lower that 3 for these rules?

Reply:
Ok, reading #7660 its clear that these events need to be stored as alerts in order to monitor and generate dashboards and so on. Perhaps we should be thinking about a new level used only for log audits, with no semantic value, in other words we are not trying to detect an attack, we are just storing audit logs from Github API

Reply:
Assigned level 3 for clone, fetch and push events.